src/routes/users.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const express = require('express');
+
+function usersRouter(router, initialUsers) {
+  let users = initialUsers;
+
+  router.get('/', (req, res) => {
+    res.send(users);
+  });
+
+  router.get('/:userId', (req, res) => {
+    const { userId } = req.params;
+    const foundUser = users.find(user => user.id === +userId);
+
+    if (!foundUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    res.send(foundUser);
+  });
+
+  router.post('/', express.json(), (req, res) => {
+    const { name } = req.body;
+    const newId = users.length ? users[users.length - 1].id + 1 : 1;
```

Comment:
You could also use new Date multiplied by a random number(this solution could be more elegant).